src/mpid/ch4/shm/posix/posix_send.h

```diff
@@ -30,53 +30,15 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_POSIX_mpi_isend(const void *buf, MPI_Aint cou
     int mpi_errno = MPI_SUCCESS;
 
     int context_offset = MPIR_PT2PT_ATTR_CONTEXT_OFFSET(attr);
-    int vsi_src, vsi_dst;
-    MPIDI_POSIX_SEND_VSIS(vsi_src, vsi_dst);
-
-    MPID_THREAD_CS_ENTER(VCI, MPIDI_VCI(vsi_src).lock);
-    mpi_errno = MPIDIG_mpi_isend(buf, count, datatype, rank, tag, comm, context_offset, addr,
-                                 vsi_src, vsi_dst, request);
-    MPID_THREAD_CS_EXIT(VCI, MPIDI_VCI(vsi_src).lock);
-
-    return mpi_errno;
-}
+    MPIR_Errflag_t errflag = MPIR_PT2PT_ATTR_GET_ERRFLAG(attr);
+    bool syncflag = MPIR_PT2PT_ATTR_GET_SYNCFLAG(attr);
 
-MPL_STATIC_INLINE_PREFIX int MPIDI_POSIX_isend_coll(const void *buf, MPI_Aint count,
-                                                    MPI_Datatype datatype, int rank, int tag,
-                                                    MPIR_Comm * comm, int attr,
-                                                    MPIDI_av_entry_t * addr,
-                                                    MPIR_Request ** request,
-                                                    MPIR_Errflag_t * errflag)
-{
-    int mpi_errno = MPI_SUCCESS;
-
-    int context_offset = MPIR_PT2PT_ATTR_CONTEXT_OFFSET(attr);
     int vsi_src, vsi_dst;
     MPIDI_POSIX_SEND_VSIS(vsi_src, vsi_dst);
 
     MPID_THREAD_CS_ENTER(VCI, MPIDI_VCI(vsi_src).lock);
-    mpi_errno = MPIDIG_isend_coll(buf, count, datatype, rank, tag, comm, context_offset, addr,
-                                  vsi_src, vsi_dst, request, errflag);
-    MPID_THREAD_CS_EXIT(VCI, MPIDI_VCI(vsi_src).lock);
-
-    return mpi_errno;
-}
-
-MPL_STATIC_INLINE_PREFIX int MPIDI_POSIX_mpi_issend(const void *buf, MPI_Aint count,
-                                                    MPI_Datatype datatype, int rank, int tag,
-                                                    MPIR_Comm * comm, int attr,
-                                                    MPIDI_av_entry_t * addr,
-                                                    MPIR_Request ** request)
-{
-    int mpi_errno = MPI_SUCCESS;
-
-    int context_offset = MPIR_PT2PT_ATTR_CONTEXT_OFFSET(attr);
-    int vsi_src, vsi_dst;
-    MPIDI_POSIX_SEND_VSIS(vsi_src, vsi_dst);
-
-    MPID_THREAD_CS_ENTER(VCI, MPIDI_VCI(vsi_src).lock);
-    mpi_errno = MPIDIG_mpi_issend(buf, count, datatype, rank, tag, comm, context_offset, addr,
-                                  vsi_src, vsi_dst, request);
+    mpi_errno = MPIDIG_mpi_isend(buf, count, datatype, rank, tag, comm, context_offset, addr,
+                                 vsi_src, vsi_dst, request, syncflag, errflag);
```

Comment:
this commit is a nice reduction. i do think we should try use `attr` in this function signature instead of continuing the grow the argument list with additional boolean parameters. this can happen separately, though.

Reply:
One argument is since the netmod/posix already decoded the `attr`, there is no need to do it again in `mpidig`, but I think it is very insignificant. Yeah, let's do that separately.

Reply:
OK, sounds good. I wouldn't mind making a small pass on some cleanup there just to get more familiar with the changes, rather than just reading them.